### PR TITLE
[google_workspace] Add support for Rules datastream and dashboard for all datastreams

### LINKED ISSUE #4079: Google Workspace | DLP audit log support
There is a need to collect the DLP alerts from the Alerts center, the api access is "https://admin.googleapis.com/admin/reports/v1/activity/users/all/applications/rules?eventName=action_complete" and to have the correct mapping to work with the alerts. 

It does not seem that any integrations exist at the moment. 

# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### LINKED ISSUE #4317: [Filebeat] [Google Workspace module] - Improve ECS utilization
We are ingesting Google Workspace data (admin, login, saml, user_accounts) into our Elasticsearch for search, detection in Elastic Security and visualization through Kibana. However, we have noticed a few specific fields where the Google Workspace module could see minor improvements in the ECS utilization.

**Note**: we are running filebeat version 8.3.3, but have noticed that none of the newer releases solves our issues.

    admin, login, saml, user_accounts

        google_workspace.kind

            ECS fields: event.kind

            Suggestion: The event.kind field is not currently populated. This should be set to the value “event”. As the document that is received could be categorized as an event.

        source.email.user

            ECS fields: source.email.user | user.email

            Suggestion: The source.email.user field is populated with the correct data. This data should also populate the user.email field.

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Added a data stream.
- Added data collection logic to the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files
- Added dashboards and visualizations.
- Added test for pipeline for the data stream.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.4.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/google_workspace directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes #4079 
- Closes #4317 
- Related https://github.com/elastic/integrations/issues/3102

## Screenshots

image
image
image
image
image
image

### CODE REVIEW

File: packages/google_workspace/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+  changes:
+    - description: Add New Rules Data Stream.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/4588
+    - description: Add Missing Dashboards.
+      type: enhancement
+      link: https://github.com/elastic/integrations/issues/3102
+    - description: Improve ECS Utilization.
+      type: enhancement
+      link: https://github.com/elastic/integrations/issues/4317
```
Comment (efd6) [inline packages/google_workspace/changelog.yml:12]: Should these issues be linked here?

File: packages/google_workspace/data_stream/rules/_dev/test/pipeline/test-common-config.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+dynamic_fields:
+  event.ingested: ".*"
```
Comment (efd6) [inline packages/google_workspace/data_stream/rules/_dev/test/pipeline/test-common-config.yml:2]: This is not needed.

File: packages/google_workspace/data_stream/rules/fields/ecs.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+- external: ecs
+  name: source.geo.continent_name
+- external: ecs
+  name: source.geo.country_iso_code
+- external: ecs
+  name: source.geo.country_name
+- name: source.geo.location
+  level: core
+  type: geo_point
+  description: Longitude and latitude.
```
Comment (efd6) [inline packages/google_workspace/data_stream/rules/fields/ecs.yml:46]:
```suggestion
- external: ecs
  name: source.geo.location
```
https://www.elastic.co/guide/en/ecs/current/ecs-geo.html#field-geo-location

File: packages/google_workspace/data_stream/rules/fields/fields.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+- name: google_workspace.rules
+  type: group
+  fields:
+    - name: actions
+      type: keyword
+      description: List of actions taken. For a list of possible values refer to https://developers.google.com/admin-sdk/reports/v1/appendix/activity/rules.
```
Comment (efd6) [inline packages/google_workspace/data_stream/rules/fields/fields.yml:6]:
```suggestion
      description: List of actions taken. For a list of possible values refer to `actions` in the [event details table]{https://developers.google.com/admin-sdk/reports/v1/appendix/activity/rules#rule_match).
```
I'd recommend similar changes for the fields below and in package-field.yml where there is a reference to the API docs as well.

File: packages/google_workspace/data_stream/rules/fields/package-fields.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          description: |
+            Name of the event. This is the specific name of the activity reported by the API. And each eventName is related to a specific Google Workspace service or feature which the API organizes into types of events.
+            For eventName request parameters in general:
+              If no eventName is given, the report returns all possible instances of an eventName.
+              When you request an eventName, the API's response returns all activities which contain that eventName. It is possible that the returned activities will have other eventName properties in addition to the one requested.
+            For more information about eventName properties, see the list of event names for various applications above in applicationName.
+        - name: type
+          type: keyword
+          description: |
+            The type of Google Workspace event, mapped from `items[].events[].type` in the original payload. Each fileset can have a different set of values for it, more details can be found at https://developers.google.com/admin-sdk/reports/v1/reference/activities/list.
```
Comment (efd6) [inline packages/google_workspace/data_stream/rules/fields/package-fields.yml:45]:
```suggestion
            The type of Google Workspace event, mapped from `items[].events[].type` in the original payload. Each fileset can have a different set of values for it, more details can be found [here](https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list#activity).
```

File: packages/google_workspace/changelog.yml (hunk 1 above)
Comment (vinit-elastic) [reply packages/google_workspace/changelog.yml:12]: Yes, we have resolved both issues in this PR.

File: packages/google_workspace/data_stream/rules/fields/fields.yml (hunk 4 above)
Comment (efd6) [reply packages/google_workspace/data_stream/rules/fields/fields.yml]: Please take a look again at this and below.

Comment (vinit-elastic) [reply packages/google_workspace/data_stream/rules/fields/fields.yml]: Hey @efd6, I have already updated this.
https://github.com/elastic/integrations/blob/9eba0eb90f1c3882f2eabf2866fcfc4ba027ebfe/packages/google_workspace/data_stream/rules/fields/fields.yml#L6
Pleases let me know if their are any other changes.

Comment (efd6) [reply packages/google_workspace/data_stream/rules/fields/fields.yml]: Oh! Sorry, UI/brain failure; the link goes to the original which doesn't have the change. Apologies.

**Review by efd6 — APPROVED** [review]

**Pull request merged**